staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -314,6 +314,10 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	terminating := apiextensionshelpers.IsCRDConditionTrue(crd, apiextensionsv1.Terminating)
+	// we want to track recently created CRDs so that in HA environments we don't have server A allow a create and server B
+	// not have observed the established, so a followup get,update,delete results in a 404. We've observed about 800ms
+	// delay in some CI environments.  Two seconds looks long enough and reasonably short for hot retriers.
+	justCreated := time.Now().Sub(apiextensionshelpers.FindCRDCondition(crd, apiextensionsv1.Established).LastTransitionTime.Time) < 2*time.Second
```

Comment:
if you only need this for create, only compute it in the create path?

Reply:
> if you only need this for create, only compute it in the create path?

Sure.